tornettools/generate_tgen.py

```diff
@@ -249,26 +249,69 @@ def __convert_privcount_key_to_tgen_key(str):
     else:
         return str
 
-def get_servers(args, n_clients):
+def __calculate_number_of_servers(args, n_public_clients, n_hs_clients):
+    # the number of servers will be a fraction of the number of clients
+    n_public_servers = n_public_clients * args.server_scale
+    n_hs_servers = n_hs_clients * args.server_scale
+    n_servers = n_public_servers + n_hs_servers
+
+    # if we don't have enough servers, scale both server counts
+    if n_servers < TGEN_SERVER_MIN_COUNT:
+        n_public_servers *= TGEN_SERVER_MIN_COUNT / n_servers
+        n_hs_servers *= TGEN_SERVER_MIN_COUNT / n_servers
+        n_servers = n_public_servers + n_hs_servers
+
+    # round after completing calculations
+    n_public_servers = round(n_public_servers)
+    n_hs_servers = round(n_hs_servers)
+    n_servers = n_public_servers + n_hs_servers
+
+    # this should still hold after rounding, but we'll double-check
+    assert n_servers >= TGEN_SERVER_MIN_COUNT
+
+    return (n_public_servers, n_hs_servers)
+
+def get_servers(args, clients):
     tgen_servers = []
 
+    n_public_clients = len([x for x in clients if not x['is_hs_client']])
+    n_hs_clients = len([x for x in clients if x['is_hs_client']])
+
+    (n_public_servers, n_hs_servers) = __calculate_number_of_servers(args, n_public_clients, n_hs_clients)
+
     # each server will be placed in a country
     # right now we use client stats to decide where to place servers
     # we may want to update this to use server-specific country distributions
     country_codes, country_probs = __load_user_data(args)
 
-    n_servers = round(n_clients * args.server_scale)
-    if n_servers < TGEN_SERVER_MIN_COUNT:
-        n_servers = TGEN_SERVER_MIN_COUNT
-    for i in range(n_servers):
+    keys = generate_onion_service_keys(args.torexe, n_hs_servers)
+
+    server_counter = 0
+
+    for i in range(n_public_servers):
+        chosen_country_code = choice(country_codes, p=country_probs)
+        server = {
+            'name': 'server{}'.format(server_counter+1),
```

Comment:
What do you think about naming these `server{}public` so that they can be grepped apart from `'server{}onionservice'`?

Reply:
Renamed the `server{}` hosts to `server{}public`.